feat(deque): Implement append and join methods with comprehensive tests

PingGuoMiaoMiao opened the pull request:
feat(deque): Add append method with tests

- Implement O(1) amortized append() for deque
- Add 9 test cases covering edge scenarios
- Document behavior in method signature

feat(deque): Optimize join method

- Replace string concatenation with Buffer
- Add type-safe generic support
- Match existing test patterns

peter-jerry-ye commented:
`join` is implemented in #2191

peter-jerry-ye commented on deque/deque.mbt:
````diff
@@ -1297,3 +1297,45 @@ pub impl[A : @json.FromJson] @json.FromJson for T[A] with from_json(json, path)
+/// ```
+pub fn[A] append(
+  self : T[A], // The deque to modify
+  elem : A // The element to append
````
Appending only one element is not consistent with `Array::append`.

PingGuoMiaoMiao commented:
> `join`[在#2191](https://github.com/moonbitlang/core/pull/2191)中实现[](https://github.com/moonbitlang/core/pull/2191)

I have deleted the join I implemented in pr
